testsupport/cleanup/clean.go

```diff
@@ -174,3 +183,34 @@ func (c *cleanTask) verifyMurDeleted(isUserSignup bool, userSignup *toolchainv1a
 	}
 	return true, nil
 }
+
+func (c *cleanTask) verifySpaceDeleted(isUserSignup bool, userSignup *toolchainv1alpha1.UserSignup, delete bool) (bool, error) {
+	// only applicable for UserSignups
+	if !isUserSignup {
+		return true, nil
+	}
+
+	space := &toolchainv1alpha1.Space{}
+	if err := c.client.Get(context.TODO(), test.NamespacedName(userSignup.GetNamespace(), userSignup.GetName()), space); err != nil {
```

Comment:
We don't use UserSignup names for Space names. 
But instead of looking for the Spaces by name, I would rather use the "creator" label. 
See: https://github.com/codeready-toolchain/host-operator/blob/master/controllers/usersignup/space.go#L11

Reply:
ok I see, thanks! it's better then using the `userSignup.Status.CompliantUsername` used here https://github.com/codeready-toolchain/host-operator/blob/7186f30efd0ad8273ed7eff02a996df35a4a5ec2/controllers/usersignup/space.go#L17 ?

Reply:
If I'm seeing it correctly seems like spaces are created in different ways during e2e tests. 
For example running this test : https://github.com/codeready-toolchain/toolchain-e2e/blob/855cd504a25d35a4a9382cfc95b9f4ab69516270/test/e2e/parallel/space_test.go#L16

The resources are creating without that expected label:

```
➜  toolchain-e2e git:(fix/test-stabilization) ✗ k get space,usersignups -o yaml
apiVersion: v1
items:
- apiVersion: toolchain.dev.openshift.com/v1alpha1
  kind: Space
  metadata:
    creationTimestamp: "2022-12-02T15:51:56Z"
    finalizers:
    - finalizer.toolchain.dev.openshift.com
    generateName: testcreatespace-
    generation: 1
    labels:
      toolchain.dev.openshift.com/appstudio-tier-hash: 28fcd1b930571badaaf23b8e10c0585e
      toolchain.dev.openshift.com/state: cluster-assigned
    name: testcreatespace-xgx9n
    namespace: toolchain-host-02123953
    ........
    
    - apiVersion: toolchain.dev.openshift.com/v1alpha1
  kind: UserSignup
  metadata:
    annotations:
      toolchain.dev.openshift.com/activation-counter: "1"
      toolchain.dev.openshift.com/last-target-cluster: member-fmuntean-sandbox-dev1.devcluster.openshift.com2
      toolchain.dev.openshift.com/skip-auto-create-space: "true"
      toolchain.dev.openshift.com/user-email: for-space-testcreatespace-xgx9n@acme.com
      toolchain.dev.openshift.com/verification-counter: "0"
    creationTimestamp: "2022-12-02T15:51:57Z"
    generation: 2
    labels:
      toolchain.dev.openshift.com/email-hash: 671a2426f9aaf785bdfefc545c0f35d2
      toolchain.dev.openshift.com/state: approved
    name: for-space-testcreatespace-xgx9n
```
but maybe it's because the signup is created using this function https://github.com/codeready-toolchain/toolchain-e2e/blob/855cd504a25d35a4a9382cfc95b9f4ab69516270/testsupport/space.go#L192

that issues the signup request with `.NoSpace()` .

So in brief , not sure it will always be able to find spaces associated with usersignups, but maybe I'm missing something.